tools/psoxy-test/README.md

```diff
@@ -1,13 +1,78 @@
 
-[Node.js](version >=16) and [npm] (version >=8).
+The primary tool is a command line interface (CLI) script that allows you to execute single "test calls" to your Worklytics Psoxy instance.  Check all the available options by running `node cli.js -h`. Those options may vary depending on whether you've deployed the Worklytics Psoxy to Amazon Web Services ([AWS]) or Google Cloud Platform ([GCP]).
 
-1. `npm i` 
-2. `node tools/psoxy-test/cli.js -h`
+## AWS
+Assuming that you've successfully deployed the Psoxy to AWS, and you've configured [Google Calendar] as data source, let's see an example:
+```shell
+node cli.js -u https://acme.lambda-url.us-east-1.on.aws/calendar/v3/calendars/primary -r arn:aws:iam::310635719553:role/PsoxyApiCaller -i user@acme.com
+```
+The `-r` option is mandatory for AWS deploys, and identifies the Amazon Resource Name (ARN) of the "role" that will be assumed (*) to be able to execute the call. The `-u` option is the URL you want to test. In this case, the URL's path matches a Google Calendar API endpoint (access the primary calendar of the currently logged-in user). The `-i` option identifies the user "to impersonate"; this option is only relevant for Google Workspace data sources.
 
+Another example for [Zoom]:
+```shell
+node cli.js -u https://acme.lambda-url.us-east-1.on.aws/v2/users -r arn:aws:iam::310635719553:role/PsoxyApiCaller
+```
+As you can see, the differences are:
+1. As this is not a Google Workspace data source, you don't need the `-i` option.
+2. The URL's path matches a [Zoom API endpoint] in this case
+
+(*) Requests to AWS API need to be [signed], so you must ensure that the machine running these scripts have the appropriate AWS credentials for the role you've selected.
+
+## GCP
+For GCP, every call must include the `-t` option, which is the identity token used for authentication (*). 
+Google Calendar example:
+```shell
+node cli.js -u https://us-central1-acme.cloudfunctions.net/psoxy-gcal/calendar/v3/calendars/primary -t <IDENTITY_TOKEN> -i user@acme.com
+```
+Zoom example:
+```shell
+node cli.js -u https://us-central1-acme.cloudfunctions.net/v2/users -t <IDENTITY_TOKEN>
+```
+
+(*) You can obtain it by running `gcloud auth print-identity-token` (using Google Cloud SDK)
+
+## Testing all endpoints for a given data source
+We also provide some npm scripts that allow you to test all the endpoints for a given data source. These scripts rely on the same Node.js module that runs the CLI script described above, but instead of providing the options via CLI, you have to create a `.env` file with the following contents:
+```shell
+# For AWS deploys
+PSOXY_URL=<url>
+PSOXY_ROLE=<role ARN>
+PSOXY_IMPERSONATE=<user email> # only for Google Workspace data sources
+PSOXY_SAVE_TO_FILE=true
+PSOXY_VERBOSE=true
+```
+```shell
+# For GCP deploys
+PSOXY_URL=<url>
+PSOXY_TOKEN=<identity token>
```

Comment:
I probably prefer just asking them to pipe it in  (showing them the right bash command to do it, eg 

```bash
node cli.js -token `gcloud auth print-identity-token` 
```

additionally, I think the tokens are only good for an hour; running `gcloud auth print-identity-token`, copying the contents, opening a flat file, pasting it in there, saving it - that's going to be really tedious. Especially for people not familiar with a terminal and/or not doing it on their local machine (such as via gcloud Cloud Shell), so that using copy-paste and a file editor is not trivial. I would say that *MOST* of our users are not savvy editing files with vim/emacs, and use IDE instead. That gets really tedious to do regularly. 

alternatively could us a real environment variable, and tell them to run
```bash
export PSOXY_TOKEN=`gcloud auth print-identity-token`
```

every hour.  That's going to be easier than editing a file.